rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp

```diff
@@ -686,14 +669,214 @@ error_code cellSysutilAvc2SetSpeakerMuting(u8 muting)
 	return CELL_OK;
 }
 
+error_code cellSysutilAvc2Load_shared(SceNpMatching2ContextId ctx_id, u32 container, vm::ptr<CellSysutilAvc2Callback> callback_func, vm::ptr<void> user_data, vm::cptr<CellSysutilAvc2InitParam> init_param)
+{
+	if (!init_param || !init_param->avc_init_param_version ||
+	    !(init_param->avc_init_param_version == 100 ||
+	     init_param->avc_init_param_version == 110 ||
+	     init_param->avc_init_param_version == 120 ||
+	     init_param->avc_init_param_version == 130 ||
+	     init_param->avc_init_param_version == 140)
+	)
+	{
+		return CELL_AVC2_ERROR_INVALID_ARGUMENT;
+	}
+
+	auto& settings = g_fxo->get<avc2_settings>();
+
+	switch (init_param->media_type)
+	{
+	case CELL_SYSUTIL_AVC2_VOICE_CHAT:
+	{
+		if (init_param->max_players < 2 ||
+		    init_param->max_players > 64 ||
+		    init_param->spu_load_average > 100 ||
+		    init_param->voice_param.voice_quality != CELL_SYSUTIL_AVC2_VOICE_QUALITY_NORMAL ||
+		    init_param->voice_param.max_speakers == 0 ||
+		    init_param->voice_param.max_speakers > 16
+		)
+		{
+			return CELL_AVC2_ERROR_INVALID_ARGUMENT;
+		}
+
+		uint32_t streaming_mode = init_param->direct_streaming_mode;
+
+		if (init_param->avc_init_param_version < 120)
+		{
+			if (init_param->avc_init_param_version >= 110)
+			{
+				switch (init_param->direct_streaming_mode)
+				{
+				case CELL_SYSUTIL_AVC2_STREAMING_MODE_NORMAL:
+				case CELL_SYSUTIL_AVC2_STREAMING_MODE_DIRECT_WAN:
+					break;
+				case CELL_SYSUTIL_AVC2_STREAMING_MODE_DIRECT_LAN:
+				default:
+					return CELL_AVC2_ERROR_INVALID_ARGUMENT;
+				}
+			}
+			else
+			{
+				streaming_mode = settings.streaming_mode;
+			}
+		}
+		else
+		{
+			switch (init_param->direct_streaming_mode)
+			{
+			case CELL_SYSUTIL_AVC2_STREAMING_MODE_NORMAL:
+				streaming_mode = CELL_SYSUTIL_AVC2_STREAMING_MODE_NORMAL;
+				break;
+			case CELL_SYSUTIL_AVC2_STREAMING_MODE_DIRECT_WAN:
+				break;
+			case CELL_SYSUTIL_AVC2_STREAMING_MODE_DIRECT_LAN:
+				if (init_param->streaming_mode.mode == CELL_SYSUTIL_AVC2_STREAMING_MODE_NORMAL)
+				{
+					settings.streaming_mode = streaming_mode;
+					return CELL_AVC2_ERROR_INVALID_ARGUMENT;
+				}
+				break;
+			default:
+				return CELL_AVC2_ERROR_INVALID_ARGUMENT;
+			}
+		}
+
+		settings.streaming_mode = streaming_mode;
+		settings.mic_out_stream_sharing = init_param->voice_param.mic_out_stream_sharing;
+
+		if (!callback_func)
+		{
+			return CELL_AVC2_ERROR_INVALID_ARGUMENT;
+		}
+
+		if (settings.avc2_cb)
+		{
+			return CELL_AVC2_ERROR_ALREADY_INITIALIZED;
+		}
+
+		settings.avc2_cb = callback_func;
+		settings.avc2_cb_arg = user_data;
+
+		// TODO
+		//return cellSysutilRegisterCallbackDispatcher(0x120, &PTR_LAB_0000552c);
```

Comment:
You can remove this line, it is just to register the callback internally.

Reply:
removed